packages/babel-preset-flow/src/normalize-options.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import findSuggestion from "levenary";
+
+const PACKAGE_NAME = "@babel/preset-flow";
+
+const TopLevelOptions = ["all", "allowDeclareFields", "ignoreExtensions"];
+
+function validateTopLevelOptions(options) {
+  for (const option of Object.keys(options)) {
+    if (!TopLevelOptions.includes(option)) {
+      const suggestion = findSuggestion(option, TopLevelOptions);
+      throw new Error(
+        `${PACKAGE_NAME}: '${option}' is not a valid top-level option.\n` +
+          `Maybe you meant to use '${suggestion}'?`,
+      );
+    }
+  }
+}
+
+function validateBoolOption(
```

Comment:
Can we make this DRYer by extracting these helper functions into a separate module?

Reply:
These helpers are derived from https://github.com/babel/babel/blob/cfaa70dcf48a0ea9a6c7adb4ae1f7925934b3933/packages/babel-preset-env/src/normalize-options.js#L139

I think we should address this in a separate PR on v7, then we can rebase this PR on that.

Reply:
Sounds good to me :)